server/src/main/java/io/druid/segment/realtime/firehose/ReplayableFirehoseFactory.java

```diff
@@ -162,6 +162,11 @@ public ReplayableFirehose(InputRowParser parser) throws IOException
               while (delegateFirehose.hasMore() && cos.getCount() < getMaxTempFileSize()) {
                 try {
                   InputRow row = delegateFirehose.nextRow();
+
+                  if (row == null) {
+                    continue;
```

Comment:
I think since we're planning to remove ReplayableFirehose in #4193, and it's currently just an implementation detail of the IndexTask, we could skip nulls here for now, and expect the class to be removed completely later.